clients/venice-push-job/src/main/java/com/linkedin/venice/hadoop/InputDataInfoProvider.java

```diff
@@ -101,13 +101,16 @@ static void loadZstdTrainingSamples(AbstractVeniceRecordReader recordReader, Pus
         continue;
       }
 
-      if (fileSampleSize + data.length > pushJobZstdConfig.getMaxBytesPerFile()) {
-        String perFileLimitErrorMsg = String.format(
-            "Read %s to build dictionary. Reached limit per file of %s.",
-            ByteUtils.generateHumanReadableByteCountString(fileSampleSize),
-            ByteUtils.generateHumanReadableByteCountString(pushJobZstdConfig.getMaxBytesPerFile()));
-        LOGGER.debug(perFileLimitErrorMsg);
-        return;
+      // At least 1 sample per file should be added until the max sample size is reached
+      if (fileSampleSize > 0) {
+        if (fileSampleSize + data.length > pushJobZstdConfig.getMaxBytesPerFile()) {
+          String perFileLimitErrorMsg = String.format(
+              "Read %s to build dictionary. Reached limit per file of %s.",
```

Comment:
nit: Since we're already touching this piece, can you replace the `String.format` with log interpolation in the `LOGGER.debug` call itself? Also, can you make the same changes for L119?
```
"Read %s to build dictionary. Reached sample limit of %s."
```

Reply:
Modified, thanks for noticing this.